### PR TITLE
Prevent inline popover to be closed when using alert provider dialogs

### PR DESCRIPTION
**Pull request by AlexandreSi**
Fixes issue mentioned in #3991.

It looks like the clicks on elements generated by the AlertProvider (ConfirmDialog, AlertDialog and ConfirmDeleteDialog) trigger the click away listener of InlinePopover.

I added a check on the click event.

### CODE REVIEW

File: newIDE/app/src/EventsSheet/InlinePopover.js [hunk 1]
```diff
@@ -54,10 +55,16 @@ export default function InlinePopover(props: Props) {
   return (
     <ClickAwayListener
       onClickAway={event => {
+        // Clicks on dialogs (or their backdrop element) generated by
+        // the AlertProvider seem to trigger this click away listener
+        // whereas they are displayed above the InlinePopover element.
```
Comment (ClementPasteau) [inline newIDE/app/src/EventsSheet/InlinePopover.js:60]: replace whereas => when

File: newIDE/app/src/UI/MaterialUISpecificUtil.js [hunk 2]
```diff
@@ -49,3 +49,22 @@ export const getMuiCheckboxValue = (element: HTMLElement): boolean => {
   }
   return true;
 };
+
+export const doesPathContainDialog = (
+  path: Array<Element>,
+  options?: { isVisible: true }
```
Comment (ClementPasteau) [inline newIDE/app/src/UI/MaterialUISpecificUtil.js:55]: What's the point of passing the options here?
Are we not always checking the dialog is visible?

File: newIDE/app/src/EventsSheet/InlinePopover.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/EventsSheet/InlinePopover.js:60]: I mean whereas so... 😅 . The inline popover is only used in the event sheet, dialogs will always be displayed above it

File: newIDE/app/src/UI/MaterialUISpecificUtil.js (hunk 2 above)
Comment (AlexandreSi) [reply newIDE/app/src/UI/MaterialUISpecificUtil.js:55]: No, the drawer is considered a dialog but it's not always visible. It's used by the in app tutorial orchestrator to know if a tooltip should be hidden or not

Comment (ClementPasteau) [reply newIDE/app/src/UI/MaterialUISpecificUtil.js:55]: I meant in the context of this function `doesPathContainDialog`. (you only use it once with isVisible true)

Comment (AlexandreSi) [reply newIDE/app/src/UI/MaterialUISpecificUtil.js]: I meant to use the same signature as for `isElementADialog` but you're right

**Review by ClementPasteau — APPROVED** [review]
nice!

**Pull request merged**